### ISSUE TITLE
BT resume download after shutdown

### ISSUE THREAD

**Issue by fridayshoes** [ziahamza/webui-aria2#513]
How do I resume a torrent after powering down my router and moving it to a new location? The partially downloaded torrent doesn't appear in the downloads list when I reboot the router and log into to webui

**Comment by no1xsyzy** [comment]
Maybe you didn't kill the process `aria2c` gently (like `kill -9`), so they are not saved in session, and you didn't tell aria 2 to save session periodically.
Just add the task again with the same target path. Torrent downloading ensures that the downloaded part are not downloaded again.

**Comment by fridayshoes** [comment]
> To recover the progress in the torrent task, just add the task again with the same torrent and the same target path. Torrent downloading ensures that downloaded parts are not downloaded again.

Yes, I was able to do this, but how do I make aria do it automatically once the router reboots?

**Comment by no1xsyzy** [comment]
Check the aria 2 manual to assure. It is an complicated question with many tweaks of aria 2.

1. Let aria 2 [`save-session`](https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-save-session).
    1. If you set [`save-session-interval`](https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-save-session-interval) to be zero (`0`), make sure you terminate `aria2c` gently. `kill` instead of `kill -9`, `sudo reboot` instead of RESET button.
2. Let aria 2 [`input-file`](https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-i)
3. Let aria 2 to [save a copy of the torrent file](https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-rpc-save-upload-metadata).
    1. If you are using Magnet URI, you need to [`bt-save-metadata`](https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-bt-save-metadata)

What's more, it is more about aria 2 itself, not the WebUI.

**Comment by fridayshoes** [comment]
> 2. Let aria 2 input-file

I'm a little stuck on this step, there's no input-file option in global settings....and if there were what do I put in place of <FILE>?

**Comment by no1xsyzy** [comment]
You should add that to your config file (included with [`conf-path`](https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-conf-path)) or command line (`aria2c --input-file=FILE`). It is reasonable that this option is only provided here since it **only** works when aria 2 starts.

**Comment by fridayshoes** [comment]
I added that line, but I started getting this error

Oh Snap! Could not connect to the aria2 RPC server. Will retry in 10 secs. You might want to check the connection settings by going to Settings > Connection Settings

Does it matter where in the config it is inserted, I put it at the end of this:

aria2c --enable-rpc --rpc-listen-all --allow-overwrite=true --auto-file-renaming=false --continue=true --dir=/mnt/USBexFAT64G/download --max-connection-per-server=4 --max-concurrent-downloads=1 --max-resume-failure-tries=10 --daemon

**Comment by no1xsyzy** [comment]
Make sure aria2c is running with `ps -elf`. If the input file does not exist, it won't create one but refuse to start. An empty file could help. I agree that this behavior is strange but it is actually intended for downloading simultaneously through the command line, not intended for sessions of RPC server.

**Comment by fridayshoes** [comment]
Thanks, but I'm now completely lost...way above my head sorry
